### PR TITLE
Add new decorator for replication lock

### PR DESCRIPTION
**Pull request by ELENAGER**
Signed-off-by: Elena Gershkovich <elenage@il.ibm.com>

### CODE REVIEW

File: controllers/servers/csi/decorators.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 from controllers.common.csi_logger import get_stdout_logger
```
Comment (matancarmeli7) [inline controllers/servers/csi/decorators.py:4]: do you know if we have unit tests for the decorators?

File: controllers/servers/csi/addons_server.py [hunk 2]
```diff
@@ -8,15 +8,16 @@
 from controllers.array_action.storage_agent import get_agent
 from controllers.common.csi_logger import get_stdout_logger
 from controllers.servers import utils
-from controllers.servers.csi.decorators import csi_method
+from controllers.servers.csi.decorators import csi_method, csi_replication_method
```
Comment (ArbelNathan) [inline controllers/servers/csi/addons_server.py:11]: are we using csi_method?

File: controllers/servers/csi/decorators.py [hunk 3]
```diff
@@ -14,15 +15,38 @@ def csi_method(error_response_type, lock_request_attribute=''):
...
-            return handle_exception(ex, context, grpc.StatusCode.ABORTED, error_response_type)
-        logger.info("finished {}".format(controller_method_name))
-        return response
+        return _set_sync_lock(lock_id, lock_request_attribute, error_response_type,
+                              controller_method, servicer, request, context)
 
     return call_csi_method
+
+
+def csi_replication_method(error_response_type, lock_replication_request_attribute=''):
```
Comment (ArbelNathan) [inline controllers/servers/csi/decorators.py:24]: why do we need `lock_replication_request_attribute` if its always the same?

File: controllers/servers/csi/decorators.py (hunk 1 above)
Comment (ELENAGER) [reply controllers/servers/csi/decorators.py:4]: Yes
    def test_disable_replication_already_processing(self):
        self._test_request_already_processing("replication_source", self.request.volume_id)

File: controllers/servers/csi/addons_server.py (hunk 2 above)
Comment (ELENAGER) [reply controllers/servers/csi/addons_server.py]: Not here, you are right. Removed

**Review by ArbelNathan — APPROVED** [review]
LGTM

**Pull request merged**